packages/ssi-sdk-core/src/utils/image.ts

```diff
@@ -1,25 +1,67 @@
 import fetch from 'cross-fetch'
-import sizeOf from 'image-size'
-import { ISizeCalculationResult } from 'image-size/dist/types/interface'
+import { imageSize } from 'image-size'
 import { IImageDimensions, IImageResource } from '../types'
 
+type SizeCalculationResult = {
+  width?: number
+  height?: number
+  orientation?: number
+  type?: string
+}
+
+const uint8ArrayToString = (uint8Array: Uint8Array): string => {
+  const decoder = new TextDecoder('utf-8')
+  return decoder.decode(uint8Array)
+}
+
+// TODO: here we're handling svg separately, remove this section when image-size starts supporting it in version 2
+const isSvg = (uint8Array: Uint8Array): boolean => {
+  const text = uint8ArrayToString(uint8Array)
+  const normalizedText = text.trim().toLowerCase()
+  return normalizedText.startsWith('<svg') || normalizedText.startsWith('<?xml')
+}
+
+const getSvgDimensions = (uint8Array: Uint8Array) => {
+  const svgContent = uint8ArrayToString(uint8Array)
+  const widthMatch = svgContent.match(/width="([^"]+)"/)
+  const heightMatch = svgContent.match(/height="([^"]+)"/)
+  const viewBoxMatch = svgContent.match(/viewBox="([^"]+)"/)
+
+  let width, height
+
+  if (widthMatch) {
+    width = widthMatch[1]
+  }
+
+  if (heightMatch) {
+    height = heightMatch[1]
+  }
+
+  if ((!width || !height) && viewBoxMatch) {
+    const parts = viewBoxMatch[1].split(' ').map(Number)
+    if (!width && parts.length === 4) {
+      width = parts[2].toString()
+    }
+    if (!height && parts.length === 4) {
+      height = parts[3].toString()
+    }
+  }
+
+  return { width: Number(width), height: Number(height), type: 'svg' }
+}
+
 export const getImageMediaType = async (base64: string): Promise<string | undefined> => {
   const buffer: Buffer = Buffer.from(base64, 'base64')
-  const result: ISizeCalculationResult = sizeOf(buffer)
-
-  switch (result.type) {
-    case undefined:
-      return undefined
-    case 'svg':
-      return `image/${result.type}+xml`
-    default:
-      return `image/${result.type}`
+  if (isSvg(buffer)) {
+    return `image/svg+xml`
   }
+  const result: SizeCalculationResult = imageSize(buffer)
+  return `image/${result.type}`
 }
 
 export const getImageDimensions = async (base64: string): Promise<IImageDimensions> => {
   const buffer: Buffer = Buffer.from(base64, 'base64')
-  const dimensions: ISizeCalculationResult = sizeOf(buffer)
+  const dimensions: SizeCalculationResult = isSvg(buffer) ? getSvgDimensions(buffer) : imageSize(buffer)
```

Comment:
Just create a getImageDimensions function, so other places can also reuse it

Reply:
we already have `getImageDimensions` if the image is svg, we're calling a special method called `getSvgDimensions` which is not exported. Do you want this to be also exported?

Reply:
Okay then it does make sense to also add an uint8array next to the string for this function

Reply:
now it's accepting both base64 string and uint8array. also added explanation at the top of the method

Reply:
also applied this to the `getImageMediaType` function. now it's also accepting `uint8array` type